### PR TITLE
Reduce the size of FabricInfo by reordering some members.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
On 32-bit systems this changes the size from 152 bytes to 144 bytes.

On 64-bit systems this changes the size from 168 bytes to 152 bytes.

### CODE REVIEW

File: src/credentials/FabricTable.h [hunk 1]
```diff
@@ -220,18 +220,69 @@ class DLL_EXPORT FabricInfo
...
+
+            // mFabricLabel is 33 bytes and has no alignment requirements.
+            totalSize += 33;
+
+            // mFabricIndex, mVendorId, mHasExternallyOwnedOperationalKey are
+            // 1, 2, 1 bytes respectively, and doing them in that order has no
+            // alignment issues.
+            totalSize += 1 + 2 + 1;
+
+            // mOperationalKey has to be aligned on a void * alignment
```
Comment (tcarmelveilleux) [inline src/credentials/FabricTable.h:253]: This is all ABI specific. Just curious why the expected size checks? The reordering is safe without all of this. Why is this present at all?

**Review by tcarmelveilleux — APPROVED** [review]
Suggest removing all the constexpr size checks: they do not significantly add,  but cause much distraction

File: src/credentials/FabricTable.h (hunk 1 above)
Comment (bzbarsky-apple) [reply src/credentials/FabricTable.h:253]: The idea was to (1) verify that the size is in fact shrinking and (2) ensure that people don't accidentally regress it (esp. by adding one boolean member and making the size 8 bytes larger or something like that).

Concretely, there are some booleans I would like to add to FabricInfo, and knowing that I am not actually changing its size in the process is useful.

But I could remove this too and just test locally.

**Review by yunhanw-google — APPROVED** [review]

**Review by pidarped — APPROVED** [review]

File: src/credentials/FabricTable.h (hunk 1 above)
Comment (bzbarsky-apple) [reply src/credentials/FabricTable.h]: OK, simplified to one very simple assert about the expected padding existing, and added some comments explaining the layout instead.

**Review by jtung-apple — APPROVED** [review]

**Review by nivi-apple — APPROVED** [review]

**Pull request merged**